checker/jdk/nullness/src/java/util/concurrent/ConcurrentHashMap.java

```diff
@@ -0,0 +1,42 @@
+package java.util.concurrent;
+import org.checkerframework.dataflow.qual.Pure;
+import org.checkerframework.dataflow.qual.SideEffectFree;
+import java.util.*;
+import org.checkerframework.checker.nullness.qual.KeyFor;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class ConcurrentHashMap<K extends @NonNull Object, V extends @NonNull Object> extends AbstractMap<K, V>
+        implements ConcurrentMap<K, V>, java.io.Serializable {
+
+    private static final long serialVersionUID = 0;
+
+     public ConcurrentHashMap(int initialCapacity,float loadFactor, int concurrencyLevel) {throw new RuntimeException("skeleton method");}
+     public ConcurrentHashMap(int initialCapacity, float loadFactor) {throw new RuntimeException("skeleton method");}
+     public ConcurrentHashMap(int initialCapacity) {throw new RuntimeException("skeleton method");}
+     public ConcurrentHashMap() {throw new RuntimeException("skeleton method");}
+     @Pure public boolean isEmpty() {throw new RuntimeException("skeleton method");}
+     @Pure public int size() {throw new RuntimeException("skeleton method");}
+     @Pure public @Nullable V get(@NonNull Object key){throw new RuntimeException("skeleton method");}
+     @Pure public boolean containsKey(@NonNull Object key) {throw new RuntimeException("skeleton method");}
+     @Pure public boolean containsValue(@NonNull Object value) {throw new RuntimeException("skeleton method");}
+     @Pure public boolean contains(@NonNull Object value) {throw new RuntimeException("skeleton method");}
+     public @Nullable V put(@NonNull K key,@NonNull V value) {throw new RuntimeException("skeleton method");}

```

Comment:
This is a self-contained example:

```
import org.checkerframework.checker.nullness.qual.NonNull;
import org.checkerframework.checker.nullness.qual.Nullable;

class AnnoKV<T extends Object> {
  void foo(T p) {}
  void bar(@NonNull T p) {}

  void baz(AnnoKV<@Nullable String> p) {
    p.foo(null);
    p.bar(null);
  }
}
```

You get a warning both for the parameter and for the invocation of bar, but not for the invocation of foo.

This is intended behavior: you are notified that the instantiation is invalid, but then the declared type argument is used for checks. To get the error we would have to change the type from the invalid instantiation to something that respects the bound - which might not be easy to do.

Having the additional `@NonNull` on the parameters gives the extra warnings, but I don't think they are useful. They clutter the annotated JDK.